### PR TITLE
`halo2 0.1.0-beta.2` etc.

### LINKED ISSUE #404: Publish `halo2_proofs 0.1.0-beta.2`

### LINKED ISSUE #458: Publish `halo2_gadgets v0.1.0-beta.1`

### PR DESCRIPTION
**Pull request by str4d**
Closes #404.
Closes #458.

### CODE REVIEW

File: halo2/Cargo.toml [hunk 1]
```diff
@@ -1,21 +1,21 @@
 [package]
 name = "halo2"
-version = "0.1.0-beta.1"
+version = "0.1.0-beta.2"
 authors = [
-    "Sean Bowe <sean@electriccoin.co>",
-    "Ying Tong Lai <yingtong@electriccoin.co>",
-    "Daira Hopwood <daira@electriccoin.co>",
```
Comment (str4d) [inline halo2/Cargo.toml:7]: This change is to reflect the contents of the `halo2` crate as of this release, which is now brand-new and only has lines from me. We can re-update it as new code is introduced after this release.

File: halo2_proofs/CHANGELOG.md [hunk 2]
```diff
@@ -6,54 +6,95 @@ and this project adheres to Rust's notion of
 [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+
+## [0.1.0-beta.2] - 2022-02-14
```
Comment (str4d) [inline halo2_proofs/CHANGELOG.md:10]: Some of the changes below are just readability refactors, but I also added extra changes that I extracted from the diff between [`halo2 0.1.0-beta.1` and current `main`](https://github.com/zcash/halo2/compare/3850b9ccd540fa38d7da4eec64ce7bd869898b42..247cd620eeb434e7f86dc8579774b30384ae02b2).

**Review by daira — APPROVED** [review]
utACK

**Review by therealyingtong — APPROVED** [review]
utACK

**Pull request merged**